### PR TITLE
Updates to camera tests to support different delta times and some further tidy-up

### PR DESCRIPTION
**Pull request by hultonha**
Note: Added a parameterized test for deltaTimes, also refactored to support having multiple tests easier (for another PR, didnt' want this follow-up one to get too big).

This is a follow-up to https://github.com/o3de/o3de/pull/3210

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Comment by AMZN-alexpete** [comment]
Thanks for implementing this! I do think this is a good change to have, though what I meant by my original suggestion was to modify the delta time every "tick" to better test fluctuating conditions a user would experience in some level where they may be looking at the sky one second with a great framerate, then as they look around, more complex geometry comes into view and the framerate tanks, or there are spikes due to asset processing, streaming, particle effects etc.

So in the test you could have an array of deltatimes like:
vector<float> deltaTimes = {generate iterationsPerDiagonal entries using a sin wave or some repeatable noise fluctuating between 5 and 100 }
```
            for (int i = 0; i < iterationsPerDiagonal; ++i)
            {
                ...
                m_controllerList->UpdateViewport({ TestViewportId, AzFramework::FloatSeconds(deltaTimes[i]), AZ::ScriptTimePoint() });
```

**Comment by hultonha** [comment]
I see got you, yup I can add this too (and I have another test related to the cursor capture stuff that's incoming). I'll see if I can update this PR before the next one otherwise this could stand on its own and I'll include the variable deltaTime in the next PR

**Comment by hultonha** [comment]
[==========] Running 4 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 1 test from ModularViewportCameraControllerFixture
[ RUN      ] ModularViewportCameraControllerFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithVaryingDeltaTime
[       OK ] ModularViewportCameraControllerFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithVaryingDeltaTime (253 ms)
[----------] 1 test from ModularViewportCameraControllerFixture (254 ms total)

[----------] 3 tests from All/ModularViewportCameraControllerDeltaTimeParamFixture
[ RUN      ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/0
[       OK ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/0 (238 ms)
[ RUN      ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/1
[       OK ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/1 (225 ms)
[ RUN      ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/2
[       OK ] All/ModularViewportCameraControllerDeltaTimeParamFixture.MouseMovementDoesNotAccumulateExcessiveDriftInModularViewportCameraWithFixedDeltaTime/2 (206 ms)
[----------] 3 tests from All/ModularViewportCameraControllerDeltaTimeParamFixture (673 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 2 test cases ran. (932 ms total)
[  PASSED  ] 4 tests.

File: Code/Editor/Lib/Tests/test_ModularViewportCameraController.cpp [hunk 1]
```diff
@@ -19,41 +19,43 @@ namespace UnitTest
...
         {
-            m_inputChannelMapper.reset();
-
-            m_controllerList->UnregisterViewportContext(TestViewportId);
-            m_controllerList.reset();
-            m_rootWidget.reset();
-
-            AllocatorsTestFixture::TearDown();
+            AzToolsFramework::ViewportInteraction::ViewportMouseCursorRequestBus::Handler::BusDisconnect();
         }
```
Comment (amzn-sean) [inline Code/Editor/Lib/Tests/test_ModularViewportCameraController.cpp:34]: question: should ```m_inputChannelMapper```  still be reset here? its set on connect, but not sure if the reset is required.

**Review by amzn-sean — APPROVED** [review]
LGTM

**Review by moraaar — APPROVED** [review]

File: Code/Editor/Lib/Tests/test_ModularViewportCameraController.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Editor/Lib/Tests/test_ModularViewportCameraController.cpp:34]: Ah yes it should, I think it's just moved - it is now destroyed in `TearDown` and is created in `SetUp`

**Pull request merged**